openrank/apiserver/models.py

```diff
@@ -1,126 +1,84 @@
-
-'''
-
-    role = models.OneToOneField()  # A Role can oprionally be assigned to a User. But a User MUST be assigned to SOME ROLE
-	name  = models.CharField(max_length=60)
-	email = models.EmailField(dmax_length=60)
-	password = models.CharField(max_length=50)
-	phone_number = models.charField(max_length=10)
-
-
-class Enrollment(models.Model)
-    user_id =    models.ForeignKey(User)
+class Enrollment(models.Model):
+    user_id = models.ForeignKey(User)
     contest_id = models.ForeignKey(Contest,on_delete=models.CASCADE)
-    marksobtained = models.IntegerField()
-    userDetails = models.charField()
-
-class Role(models.Model):
-	role = models.charField()
-
-
-class Question(models.Model):
-
-	description = models.TextField()
-	total_marks = models.IntegerField()
-	tags = models.manyToMany(Tag)
-
-	 
+    marks_obtained = models.IntegerField()
+    user_details = models.CharField(max_length=100) #chesinaka already ni!!!
 
+class Tag(models.Model):
+    tag_description =   models.CharField()     
+	
+class Question(models.Model):                      
+    description = models.TextField()            
+    total_marks = models.IntegerField()
+    tags = models.ManyToManyField(Tag)               
 
 class ContestQuestion(models.Model):
-	question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
-	contest_id = models.ForeignKey(Contest,on_delete=models.CASCADE)
-	total_marks  = models.IntegerField()
-	time_limit   = models.IntegerField()
-	memory_limit = models.IntegerField()  
-	variable_constraints = models.CharField(max_length=45)
-
+    question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
+    contest_id = models.ForeignKey(Contest,on_delete=models.CASCADE)
+    total_marks  = models.IntegerField()
+    time_limit   = models.IntegerField()
+    memory_limit = models.IntegerField()  
+    variable_constraints = models.CharField(max_length=100)
 
 class Language(models.Model):
-	language = models.charField()
-
+    language = models.CharField(max_length = 100)  
 
 class QuestionLanguage(models.Model):
-	question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
-	language_id = models.ForeignKey(Language,on_delete=models.CASCADE)
-	function_signature =  models.TextField()
-	function_parameters = models.charField()
-	starter_code =        models.charField()
-
+    question_id = models.ForeignKey(Question,on_delete=models.CASCADE)
+    language_id = models.ForeignKey(Language,on_delete=models.CASCADE)
+    function_signature =  models.TextField()
+    function_parameters = models.CharField(max_length=100)
+    starter_code = models.CharField(max_length=100)
 
 class Tag(models.Model):
-	tag_description =   models.charField()  #A charField
-	questions = models.manyToMany(Question)
-    
+    tag_description = models.CharField(max_length=100)     
+    questions = models.ManyToManyField(Question)
 
+class TestCaseBatch(models.Model):
+    marks = models.IntegerField()
+    title = models.CharField()	
```

Comment:
Not important, but you can set limit here as the title will always be single line